### PR TITLE
Docker: add installer.sh

### PR DESCRIPTION
**Pull request by ShockDW**
This pull request is meant to provide a user friendly install.sh for deploying Szurubooru.

Features:
- Prompts user for config values, then deploys Szurubooru automatically
- Performs the necessary ownership change of the local /data/ mount point to avoid problems uploading images #216 
- Checks host architecture and automates the changes outlined in readme.md and kaijuagenda's rpi3 fork for ARM hosts

I'm a big fan of documentation, so I tried my best to detail the purpose of the various operations and functions via comments. Naturally, these can be changed or removed if found to be too verbose or unclear.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+#
+# * Modifies docker-compose.yml rather than relying on a static alternative file to hopefully
+#   avoid the need for parallel maintenance of two files.
+#############################################################################################
+
+# Check for root to ensure the accessibility of the Docker daemon, and to 
+# allow mount point permissions to be set after building.
+
+if [[ $EUID -ne 0 ]]; then
+    echo "Sorry, but this script needs root to properly function.  Please try again using sudo ./install.sh" 1>&2
```
Comment (sgsunder) [inline install.sh:19]: You don't always need root to run docker, see:

https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+########################################################################################
+
+if [[ "$(uname -m)" == 'a'* ]]; then
+    echo "ARM architecture detected.  Modfying docker-compose.yml accordingly."
+    sed -zi "s|image: szurubooru/server:latest|build:\n      context: ./server|; \     # Modify services/server to build locally instead of pulling from DockerHub
+             s|image: szurubooru/client:latest|build:\n      context: ./client|; \     # Modify services/client to build locally instead of pulling from DockerHub
+             s|depends_on:\n      - sql|depends_on:\n      - sql\n      - elasticsearch|; \ # Add dependency for alternate arm compatible elasticsearch defined below.
+             s|POSTGRES_HOST: sql|POSTGRES_HOST: sql\n      ESEARCH_HOST: elasticsearch|" \ # Point Postgres to the alternate elasticsearch container
+            docker-compose.yml
+    # "Add a new section to define the alternate elasticsearch service
```
Comment (sgsunder) [inline install.sh:47]: You can drop this part, we haven't used elasticsearch in quite a while

**Review by sgsunder — CHANGES_REQUESTED** [review]

**Comment by sgsunder** [comment]
Thank you! This will probably help new users a lot

File: install.sh (hunk 1 above)
Comment (ShockDW) [reply install.sh]: Good point!  Removed Root check.

File: install.sh (hunk 2 above)
Comment (ShockDW) [reply install.sh]: Removed. :)

**Comment by ShockDW** [comment]
My pleasure!  I'm fairly new to Python, so I'm just glad to have found a way to contribute.  Hopefully I can pitch in some more in the future.  

All suggested changes have been made, along with a few updates to output as well as a few fixes for flaws I noticed while testing said changes.  Please let me know if you notice anything else out of place, and thanks for the review!

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+#
+# * Modifies docker-compose.yml rather than relying on a static alternative file to hopefully
+#   avoid the need for parallel maintenance of two files.
+#############################################################################################
+
+echo "Welcome to Szurubooru.  This install script will ask you to"
+echo "set a few config values, then get szurubooru up and running."
+
+# Check to ensure Docker and Docker-Compose are both installed before proceeding
+if ! command -v docker &> /dev/null || ! command -v docker-compose &> /dev/null; then
```
Comment (Jazzepi) [inline install.sh:19]: id -Gn | grep -i docker
^ will return 0 when the currently executing user in the docker group. You can unfortunately successfully run `docker` even if you don't have access to the socket it's talking to.

Additionally you could run `docker ps` to see if they can actually talk to their docker socket. It'll return 1 if they can't.

Comment (ShockDW) [reply install.sh:26]: Excellent information!

This section was only really meant to serve as an install check for Docker and Docker-Compose were installed, but I've added an additional check on `docker ps`'s  return status to ensure the current user can access the daemon/socket.  Thanks Jazzepi!

Comment (Jazzepi) [reply install.sh:26]: No problem :) I love what you did with the script, so kudos on that.

**Review by sgsunder — APPROVED** [review]

**Comment by sgsunder** [comment]
I'll merge this in, but I think I will make further refinements later on to streamline this

**Pull request merged**